rails/spec/models/place_spec.rb

```diff
@@ -6,148 +6,115 @@
-          @place = described_class.last
-          @place.photo.blob.content_type = "file/png"
-          @place.photo_format
-        end
-
-        it { expect(@place.photo.blob.content_type.start_with?('image/')).to be(false) }
-        it { expect(@place.errors.count).to eq(1) }
+      it "successfully imports places with name audio" do
+        described_class.import(
+          file_fixture('place_with_media.csv'),
+          community.id,
+          mapped_headers.merge({name_audio: "audio"})
+        )
+        expect(described_class.last.name_audio).to be_attached
       end
     end
 
-    context 'when the attachment is located in an image folder' do
-      describe 'should not add an error' do
-        before do
-          @fixture_data = file_fixture('place_with_media.csv').read
-          described_class.import_csv(@fixture_data, community)
-        end
-        let!(:place) { described_class.last }
-
-        it { expect(place.photo.blob.content_type.start_with?('image/')).to be(true) }
-        it { expect(place.errors.count).to eq(0) }
+    context "when CSV has duplicate rows" do
+      it "imports first row, returns rest as an array rows skipped" do
+        import = described_class.import(file_fixture('duplicated_places.csv'), community.id, mapped_headers)
+        expect(import[:successful]).to eq(1)
+        expect(import[:skipped_rows].length).to eq(1)
       end
     end
-  end
-
-  describe 'photo_url' do
-    describe 'should return a url path' do
-      let(:community) { FactoryBot.create(:community) }
 
+    context "when Community already has a Place with the same name" do
       before do
-        @fixture_data = file_fixture('place_with_media.csv').read
-        described_class.import_csv(@fixture_data, community)
+        FactoryBot.create(:place, name: "Iacitata", community: community)
+      end
+      it "does not add duplicate row, returns in duplicate array" do
+        import = described_class.import(file_fixture('place_without_media.csv'), community.id, mapped_headers)
+        expect(import[:successful]).to eq(0)
+        expect(import[:duplicated_rows].length).to eq(1)
       end
-      let!(:place) { described_class.last }
-      let!(:file_name) { place.photo.filename.to_s }
-
-      it { expect(place.photo_url).to be_truthy }
-      it { expect(place.photo_url.include?(file_name)).to be(true) }
     end
-  end
 
-  describe 'point_geojson' do
-    let(:community) { FactoryBot.create(:community) }
-
-    it 'should return a geoJson point' do
-      fixture_data = file_fixture('place_with_media.csv').read
-      described_class.import_csv(fixture_data, community)
-      place = described_class.first
-      expect(place.point_geojson.keys).to include('properties')
+    context "when row is invalid" do
+      it "returns invalid rows with errors" do
+        import = described_class.import(file_fixture('invalid places.csv'), community.id, mapped_headers)
```

Comment:
looking at this fixture file, I'm not sure why it's invalid. Could you tell me why? 

EDIT: Oh, is it because the long/lat are out of range?

Reply:
> EDIT: Oh, is it because the long/lat are out of range?

Yup!